libraries/shared/storage/diff_repository_test.go

```diff
@@ -133,6 +133,21 @@ var _ = Describe("Storage diffs repository", func() {
 			Expect(count).To(Equal(1))
 		})
 
+		It("does duplicate storage value if same value only exists at a later block", func() {
+			_, createErr := repo.CreateStorageDiff(fakeStorageDiff)
+			Expect(createErr).NotTo(HaveOccurred())
+
+			duplicateDiff := fakeStorageDiff
+			duplicateDiff.BlockHeight = fakeStorageDiff.BlockHeight - 1
+			createTwoErr := repo.CreateBackFilledStorageValue(duplicateDiff)
+			Expect(createTwoErr).NotTo(HaveOccurred())
+
+			var count int
+			getErr := db.Get(&count, `SELECT count(*) FROM public.storage_diff`)
+			Expect(getErr).NotTo(HaveOccurred())
+			Expect(count).To(Equal(2))
+		})
```

Comment:
Just to make sure I'm understanding this scenario correctly is this what it's capturing?

block 2 - storage slot value is set to 200
block 3 - storage slot value is set to 1

And then we're trying to insert a storage value at block 1, with a storage value of 1 also.

Reply:
I think it's more like:

block 2 - storage slot value is set to 1

And then we're trying insert a storage value at block 1, with a storage value of 1 also.

But the purpose of doing so is basically to avoid the scenario you're describing - where we fail to insert a value at an early block just because it happens to have had the same value at a later block.

It does mean that we create the possibility of duplicates if the backfill process is run out of order, but I'm not sure attempting to avoid that is worth the additional complexity (I think we'd need to scan for the closest row in both directions to check for duplicates, and do so on the assumption that we have all relevant rows in both directions).